### PR TITLE
Offline flag for v0.12

### PR DESCRIPTION
**Pull request by yukinying**
This is based from #72 to add `--offline` flag. It also makes unit test possible without connecting to GCP API. 

This pull request would invalidate #38 as #38 has no version switch logic.

### CODE REVIEW

File: converters/google/convert.go [hunk 1]
```diff
@@ -86,13 +86,15 @@ type AssetResource struct {
...
-func NewConverter(resourceManager *cloudresourcemanager.Service, project, ancestry, credentials string) (*Converter, error) {
+func NewConverter(resourceManager *cloudresourcemanager.Service, project, ancestry, credentials string, offline bool) (*Converter, error) {
 	cfg := &converter.Config{
 		Project:     project,
 		Credentials: credentials,
 	}
-	if err := cfg.LoadAndValidate(); err != nil {
-		return nil, errors.Wrap(err, "configuring")
+	if !offline {
+		if err := cfg.LoadAndValidate(); err != nil {
```
Comment (morgante) [inline converters/google/convert.go]: Is it safe to skip the loading step entirely in offline mode?

File: converters/google/convert.go [hunk 2]
```diff
@@ -122,6 +124,7 @@ type Converter struct {
 
 	cfg *converter.Config
 
+	// Talk to GCP resource manager. This field would be nil in offline mode.
```
Comment (morgante) [inline converters/google/convert.go]: I think we should make Converter into an interface and add offline and online implementations.

**Comment by morgante** [comment]
Overall looks fine, but I concur with @briantkennedy that we should change Converter to an interface with offline/online implementations.

**Comment by yukinying** [comment]
I'd prefer doing that on a separate pull request (with an issue tracker). How does that sound?

**Comment by morgante** [comment]
Ok, opened https://github.com/GoogleCloudPlatform/terraform-validator/issues/75 to track.

**Pull request merged**